modules/core/include/visp3/core/vpArray2D.h

```diff
@@ -172,7 +172,7 @@ template <class Type> class vpArray2D
   \param c : Array number of columns.
   */
   vpArray2D<Type>(unsigned int r, unsigned int c) :
-  #ifdef VISP_HAVE_CXX11
+  #if defined(VISP_HAVE_CXX11) || defined(VISP_HAVE_CXX14)
```

Comment:
I am wondering if it is not possible to have something like [`__cplusplus`](https://stackoverflow.com/questions/2324658/how-to-determine-the-version-of-the-c-standard-used-by-the-compiler/7132549#7132549). This way the check could be just:
- `VISP_CXX_VERSION >= 201103L` (C++11)
- `VISP_CXX_VERSION >= 201703L` (C++17)

Reply:
May be possible for linux, but I'm not sure that it works with msvc (see [cxx11.cpp](https://github.com/lagadic/visp/blob/master/cmake/checks/cxx11.cpp))

Reply:
@s-trinh I agree that it would simplify the code and allow to consider all the standard if we have something like:
```
#if (VISP_CXX_STANDARD_VERSION >= VISP_CXX_STANDARD_11)
```
rather than 
```
#if defined(VISP_HAVE_CXX11) || defined(VISP_HAVE_CXX14) || defined(VISP_HAVE_CXX17) || defined(VISP_HAVE_CXX2a) 
```
But the question is which values for `VISP_CXX_STANDARD_VERSION`:
- we cannot use `98`, `11`, `14`, `17`, `2a`
- what do you think about hexa values like: `0x1998`, `0x2011`, `0x2014`, `0x2017`, `0x202a` ?

Reply:
I would used the same values than `__cplusplus`: `201103L`, `201402L`, `201703L`.

Reply:
Perfect. You are right